lib/main.js

```diff
@@ -1,37 +1,45 @@
 'use strict';
 
-var autoprefixer = require('autoprefixer');
 var cheerio = require('cheerio');
+var autoprefixer = require('autoprefixer');
+
 var VERSION = require('../package.json').version;
 
-var HtmlAutoprefixer = function() {
+var HTMLAutoprefixer = function() {
   this.version = VERSION;
+  this.re = /#tmp\d+{(.+?)}/g;
   return this;
 };
 
-HtmlAutoprefixer.prototype.process = function(htmlString, cheerioOpts) {
+HTMLAutoprefixer.prototype.process = function(htmlString, cheerioOpts, autoprefixerOpts) {
   var $ = cheerio.load(htmlString, cheerioOpts);
   var styles = $('style').contents();
-  var elementsWithStyleTag = $('[style]');
+  var elementsWithStyleAttr = $('[style]');
   var prefixed;
 
   styles.each(function(index, style) {
-    var css = style.data;
-    prefixed = autoprefixer.process(css).css;
+    prefixed = autoprefixer.process(style.data, autoprefixerOpts).css;
     style.data = prefixed;
   });
 
-  elementsWithStyleTag.each(function(index, element) {
-    var css = '.tmp{' + element.attribs.style + '}';
-    try {
-      prefixed = autoprefixer.process(css).css;
-      element.attribs.style = prefixed.replace('.tmp{', '').replace('}', '');
-    } catch (e) {
-      throw('Error parsing: element: ' + element.name + ' style: ' + element.attribs.style);
-    }
-  });
+  if (elementsWithStyleAttr.length > 0) {
+    var tmp = '';
+    elementsWithStyleAttr.each(function(index, element) {
+      tmp += '#tmp' + index + '{' + element.attribs.style + '}';
+    });
+
+    prefixed = autoprefixer.process(tmp, autoprefixerOpts).css;
+
+    var re = this.re;
+    elementsWithStyleAttr.each(function(index, element) {
+      var match = re.exec(prefixed);
```

Comment:
I never have used `exec` before. Very cool how you can iterate that way over multiple matches.